mvg/mvg.py

```diff
@@ -533,11 +539,32 @@ def list_measurements(self, sid: str) -> list:
         """
         logger.info("endpoint %s", self.endpoint)
         logger.info("retrieving all measurements from source id=%s", sid)
-
-        response = self._request("get", f"/sources/{sid}/measurements")
-        all_measurements = response.json()
-
-        logger.info("%s measurements in database", len(all_measurements))
+        url = f"/sources/{sid}/measurements"
+        if offset is None and limit is None:
+            # List all by default
+            response = self._request("get", url)
+            resp_first = response.json()
+            all_measurements = resp_first["items"]
+            num_measurements = resp_first["total"]
+            limit = resp_first["limit"]
+            num_reqs = (num_measurements - 1) // limit
+            for idx in range(1, num_reqs + 1):
+                offset = idx * limit
+                response = self._request("get", url, params={"offset": offset})
+                all_measurements += response.json()["items"]
+        else:
+            params = {}
+            if limit is not None:
+                params["limit"] = limit
+            if offset is not None:
+                params["offset"] = offset
```

Comment:
A suggestion to reduce the level of nested if statements is to move the declaration of params to the beginning:

```python
    params = {}
    if limit is not None:
        params["limit"] = limit
    if offset is not None:
        params["offset"] = offset

    if not params:
        # List all by default
    else:
        # Use pagination
```